tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -1224,6 +1224,92 @@ bool FindMatMulBiasAddAndGelu(RemapperContext* ctx, int node_index,
   return (found_gelu_exact || found_gelu_approximate);
 }
 
+bool FindMulAndMaximum(RemapperContext* ctx, int node_index,
+                  std::map<string, int>* matched_nodes_map,
+                  std::set<int>* remove_node_indices) {
+  using utils::MatchingDirection;
+  using utils::NodeStatus;
+
+  // Convert Mul+Maximum to LeakyRelu
+  // maximum(x, alpha * x) = LeakyRelu(x)
+  utils::OpTypePattern mulmax_pattern{
+    "Maximum", "max_to_leakyrelu", NodeStatus::kReplace,
+    {
+      { "Mul", "mul", NodeStatus::kRemove,
+        {
+          { "*", "input", NodeStatus::kRemain},
+          { "Const|Cast", "alpha", NodeStatus::kRemain}
+        }
+      },
+      { "*", "input", NodeStatus::kRemain}
+    }
+  };
+
+  // Check for allowed datatypes
+  auto* max_node_def = ctx->graph_view.GetNode(node_index)->node();
+  if (!HasDataType(max_node_def, DT_HALF) &&
+      !HasDataType(max_node_def, DT_BFLOAT16) &&
+      !HasDataType(max_node_def, DT_FLOAT) &&
+      !HasDataType(max_node_def, DT_DOUBLE))
+    return false;
+
+  // Current implementation has support only
+  // for CPU when oneDNN is enabled.
+  // TODO(intel-tf): This will be removed when fully tested with GPU
+  if (!NodeIsOnCpu(max_node_def) && !IsMKLEnabled())
+    return false;
+
+  bool found_op_type_match = false;
+  utils::SubGraphMatcher<MatchingDirection::kFollowInputs> graph_matcher(
+      &(ctx->graph_view));
+  matched_nodes_map->clear();
+  remove_node_indices->clear();
+
+  found_op_type_match = graph_matcher.GetMatchedNodes(
+      mulmax_pattern, {}, ctx->graph_view.GetNode(node_index),
+      matched_nodes_map, remove_node_indices);
+
+  // Check if the value of alpha >= 0 as required for LeakyRelu
+  if (found_op_type_match) {
+    const auto* alpha_node_view =
+      ctx->graph_view.GetNode(matched_nodes_map->at("alpha"));
+    const auto* alpha_node_def = alpha_node_view->node();
+
+    float alpha_val;
+    if (alpha_node_def->op() == "Cast") {
+      const auto& regular_fanin_0 = alpha_node_view->GetRegularFanin(0);
+      const auto* regular_node_view = regular_fanin_0.node_view();
+      const auto* const_node = regular_node_view->node();
+      if (const_node != nullptr && const_node->op() == "Const") {
+        Tensor alpha_tensor;
+        alpha_tensor.FromProto(const_node->attr().at("value").tensor());
```

Comment:
Please fix:
```
error: ignoring return value of function declared with 'warn_unused_result' attribute [-Werror,-Wunused-result]
        alpha_tensor.FromProto(const_node->attr().at("value").tensor());
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
@penpornk I can't seem to reproduce this error locally. Can you please provide me with the config that you used that generates this error? Or point me to a log where you found it? Thanks!

Reply:
I can't reproduce it in OSS either. It's probably because gcc / clang difference. (Internally we use clang.) Although I don't see why gcc shouldn't be able to detect this. `FromProto` is explicitly tagged with `TF_MUST_USE_RESULT` which also expands to `__attribute__((warn_unused_result))` for gcc.

Can you make the code check the return value from `FromProto` and proceed accordingly? (E.g., should `FindMulAndMaximum` return `false` of `FromProto` fails to parse?)

Reply:
Thanks for the feedback. I made changes as requested, please let me know if the error still persists.